src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -134,15 +135,18 @@
         extends ZoneId
         implements TemporalAccessor, TemporalAdjuster, Comparable<ZoneOffset>, Serializable {
 
-    /** Cache of time-zone offset by offset in seconds. */
-    private static final ConcurrentMap<Integer, ZoneOffset> SECONDS_CACHE = new ConcurrentHashMap<>(16, 0.75f, 4);
-    /** Cache of time-zone offset by ID. */
-    private static final ConcurrentMap<String, ZoneOffset> ID_CACHE = new ConcurrentHashMap<>(16, 0.75f, 4);
-
     /**
      * The abs maximum seconds.
      */
     private static final int MAX_SECONDS = 18 * SECONDS_PER_HOUR;
+
+    // Max slot number for either direction - or +
+    private static final int MAX_SECONDS_CACHE_SLOT = MAX_SECONDS / (15 * SECONDS_PER_MINUTE);
+
+    /** Cache of time-zone offset by offset in seconds [-18h, +18h] for each even quarter of an hour. */
+    private static final AtomicReferenceArray<ZoneOffset> SECONDS_CACHE = new AtomicReferenceArray<>(MAX_SECONDS_CACHE_SLOT * 2 + 1);
```

Comment:
Can we use regular array instead?

Reply:
We can but that entails special handling to ensure thread-safety. I will provide such a solution. Thanks.

Reply:
If you need the thread-safety, perhaps sticking with `AtomicReferenceArray` is a simpler solution than the regular array one.